src/instructlab/configuration.py

```diff
@@ -36,6 +36,7 @@
 
 ILAB_PACKAGE_NAME = "instructlab"
 CONFIG_FILENAME = "config.yaml"
+CONFIG_VERSION = "1.0.0"
```

Comment:
Should we be doing the versioning here using semver, or should we instead do a discrete number like `v1`, `v2`, etc., so that we can handle migrations between configs?

Reply:
Would like to get an opinion on this from @tiran, @leseb, or @nathan-weinberg.

Reply:
I think a discrete number may make more sense here, we do something like that for the schema: https://github.com/instructlab/schema/tree/main/src/instructlab/schema

Reply:
as long as v1 can also be v1.1 or something like that I am fine with it. 

We need to have different major/minor versions of the cfg in between minor releases

Reply:
@alimaredia has a dev doc started here https://github.com/instructlab/dev-docs/pull/117 that can be added in